gcc/rust/expand/rust-macro-expand.cc

```diff
@@ -3677,21 +3672,8 @@ MacroExpander::match_n_matches (
 
   // Check if the amount of matches we got is valid: Is it more than the lower
   // bound and less than the higher bound?
-  auto result = hi_bound ? match_amount >= lo_bound && match_amount <= hi_bound
-			 : match_amount >= lo_bound;
-
-  // We can now set the amount to each fragment we matched in the substack
-  auto &stack_map = sub_stack.peek ();
-  for (auto &fragment_id : fragment_identifiers)
-    {
-      auto it = stack_map.find (fragment_id);
-
-      rust_assert (it != stack_map.end ());
-
-      it->second.set_match_amount (match_amount);
-    }
-
-  return result;
+  return hi_bound ? match_amount >= lo_bound && match_amount <= hi_bound
```

Comment:
This is just a style thing when it comes to a long ternery statement like this I would suggest change it into this:

```c
bool did_meet_hi_bound = hi_bound ? match_amount <= hi_bound : true;
bool did_meet_lo_bound = match_amound >= lo_bound;

return did_meet_hi_bound && did_meet_lo_bound
```